### PR TITLE
Finish-project

### PR DESCRIPTION
**Pull request by gilberthappi**
Use JavaScript to make websites dynamic and build basic single page apps.
Use ES6 syntax.
Use ES6 modules.
Use callbacks and promises.
Use webpack.
Apply JavaScript best practices and language style guides in code.
Use AAA pattern for unit tests.
Write units tests for a JavaScript app.
Follow Gitflow.
Solve simple git conflicts.
Send and receive data from an API.
Use API documentation.
Understand and use JSON.
Make JavaScript code asynchronous.
Perform a code review for a team member.
https://drive.google.com/drive/folders/1ErHYjeGAXM18Z9U63YcrEVbG5Z-BPxcQ?usp=sharing

### CODE REVIEW

**Comment by zeff96** [comment]
@gilberthappi, Hello, kindly accept my call on zoom

**Review by zeff96 — COMMENTED** [review]
Hi @gilberthappi and @MarkYegon7 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

invalid

You have submitted a project, that lacks descriptive title and PR summary, Kindly fix that before submitting the project again

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/modules/pageBody.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  )}">Read more & comments</button>
+        </div>`;
+      showCommentModal();
+    });
+
+    count += 1;
+  });
+
+  counter.innerText = `(${count * 2})`;
+  counter.style.color = 'red';
```
Comment (ShadyShawkat) [inline src/modules/pageBody.js:172]: - [ ] As per [this](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md#:~:text=Even%20if%20the%20API%20gives%20you%20these%20numbers%2C%20you%20will%20create%20a%20specific%20function%20to%20calculate%20these%20numbers%20in%20each%20page.%20These%20count%20functions%20need%20to%20be%20covered%20with%20unit%20tests%20using%20Jest.) requirement, please make sure that there's a function that on its own calculate the number of items on the page and then display it. This will open up the possibility of testing it as a pure function without mocking it. This also abies by the SOLID principle.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import likeMovie from "./modules/likeFeature.js";
+/* eslint-disable linebreak-style */
+/* eslint-disable max-len */
+/* eslint-disable linebreak-style */
+/* eslint-disable no-multiple-empty-lines */
+/* eslint-disable linebreak-style */
+/* eslint-disable array-callback-return */
+/* eslint-disable arrow-body-style */
+/* eslint-disable import/no-unresolved */
+/* eslint-disable quotes */
```
Comment (ShadyShawkat) [inline src/index.js:14]: - [ ] Please remove `eslint-disable` lines as that defies the purpose of using linters in the first place. (This goes to all files that `eslint-disable` is used in)

File: src/__mock__/countComment.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const comments = (commentList) => {
+  let counter;
+  commentList.forEach((comment, index) => {
+    counter = index + 1;
+  });
+  return counter;
+};
+
+module.exports = comments;
```
Comment (ShadyShawkat) [inline src/__mock__/countComment.js:9]: - [ ] The following method counts the number of elements in an array which is different than what a counter function for this project would do. You can implement the counter function as a pure separate method which can be used in the tests directly as it's a pure function.

File: src/app.test.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  test("First array test", () => {
+    expect(movies(firstTestArray)).toBe(8);
+  });
+  test("Second array test", () => {
+    expect(movies(secondTestArray)).toBe(3);
+  });
+  test("Third array test", () => {
+    expect(movies(thirdTestArray)).toBe(5);
+  });
+});
```
Comment (ShadyShawkat) [inline src/app.test.js:21]: - [ ] Kindly remember to cover all edge cases of the method. (like if there's no movies)

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @gilberthappi and @MarkYegon7, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 
- Good overall design ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 5]
```diff
@@ -1,293 +1,108 @@
...
 
-For more information on the importance of a professional README for your repositories: https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/readme_best_practices.md
--->
+## Live Demo (not avialable)
 
-<div align="center">
-  <!-- You are encouraged to replace this logo with your own! Otherwise you can also remove it. -->
-  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
-  <br/>
+Click [Here]([https://gilberthappi.github.io/Movie-Site/dist])To open the live demo
```
[the remark this answers is not in the document]
Comment (gilberthappi) [reply README.md:26]: Okey Thanks

File: src/index.js (hunk 2 above)
Comment (gilberthappi) [reply src/index.js:14]: Okey Thanks

File: src/__mock__/countComment.js (hunk 3 above)
Comment (gilberthappi) [reply src/__mock__/countComment.js:9]: okey thanks

File: src/app.test.js (hunk 4 above)
Comment (gilberthappi) [reply src/app.test.js:21]: okey thanks

**Review by Efiamotu-1 — APPROVED** [review]
Hi @gilberthappi & @MarkYegon7, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gilberthappi** [comment]
Thanks
